### PR TITLE
add profiler logging if method match fails

### PR DESCRIPTION
**Pull request by lucaspimentel**
Changes proposed in this pull request:
- in the clr profiler, add debug logging if method failed to match
  - invalid signature size
  - type count don't match
  - type names don't match

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -403,17 +403,27 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
         continue;
       }
 
-      if (method_replacement.wrapper_method.method_signature.data.size() < 3) {
+      auto wrapper_method_signature_size =
+          method_replacement.wrapper_method.method_signature.data.size();
+
+      if (wrapper_method_signature_size < 3) {
```
Comment (colin-higgins) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: This makes me realize that we should actually have a minimum of 5 with the addition of the opCode and mdToken.

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 2]
```diff
@@ -403,17 +403,27 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
...
         // This is invalid, we should always have the wrapper fully defined
         // Minimum: 0:{CallingConvention}|1:{ParamCount}|2:{ReturnType}
         // Drop out for safety
+        if (debug_logging_enabled) {
+          Debug("JITCompilationStarted skipping method: function_id=",
```
Comment (colin-higgins) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Would it be amenable to add type name and method name to this?

**Comment by colin-higgins** [comment]
Great adds, I've been caught when adding new methods as well.

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 2 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Added method tokens and names in f9b6c2ab2748f0ae1e4f579157a455d9ffb71333.

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 1 above)
Comment (lucaspimentel) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: Update this check to `< 5` in 0ba322610d681ddc0fc00a540665ae74c3b2ea10.

**Pull request merged**